services/QuillLMS/engines/comprehension/lib/comprehension/comprehension/spelling_check.rb

```diff
@@ -0,0 +1,55 @@
+module Comprehension
+  class SpellingCheck
+
+    ALL_CORRECT_FEEDBACK = 'Correct spelling!'
+    INCORRECT_FEEDBACK = 'Try again. There may be a spelling mistake.'
+    FEEDBACK_TYPE = 'spelling'
+    RESPONSE_TYPE = 'response'
+    BING_API_URL = 'https://api.cognitive.microsoft.com/bing/v7.0/SpellCheck'
+    SPELLING_CONCEPT_UID = 'H-2lrblngQAQ8_s-ctye4g'
+    attr_reader :entry
+
+    def initialize(entry)
+      @entry = entry
+    end
+
+    def feedback_object
+      return {} if error.present?
+      {
+        feedback: optimal? ? ALL_CORRECT_FEEDBACK : INCORRECT_FEEDBACK,
+        feedback_type: FEEDBACK_TYPE,
+        optimal: optimal?,
+        response_id: '',
+        entry: @entry,
+        concept_uid: SPELLING_CONCEPT_UID,
+        rule_uid: '',
+        highlight: optimal? ? [] : misspelled.map {|m| { type: RESPONSE_TYPE, id: nil, text: m['token']}}
```

Comment:
I would extract `misspelled.map {|m| { type: RESPONSE_TYPE, id: nil, text: m['token']}}` out into a private method which will _highlight_ what the map is doing